### PR TITLE
Node integration test enviroment

### PR DESCRIPTION
**Pull request by kraenhansen**
## What, How & Why?

This fixes up the Node.js integration test environment, making it ESM friendly in the process.

I had to temporarily skip the Node.js specific "analytics" tests, as they need to be migrated off the non-esm Node.js APIs.

## ☑️ ToDos
* [ ] 📝 Changelog entry
* [x] 📝 `Compatibility` label is updated or copied from previous entry
* [x] 📝 Update `COMPATIBILITY.md`
* [x] 🚦 Tests
* [x] 🔀 Executed flexible sync tests locally if modifying flexible sync
* [x] 📦 Updated internal package version in consuming `package.json`s (if updating internal packages)
* [x] 📱 Check the React Native/other sample apps work if necessary
* [x] 📝 Public documentation PR created or is not necessary
* [x] 💥 `Breaking` label has been applied or is not necessary

### CODE REVIEW

File: integration-tests/tests/src/index.ts [hunk 1]
```diff
@@ -27,8 +27,8 @@ afterEach(() => {
   }
 });
 
-import "./utils/import-app.test";
-import "./utils/chai-plugin.test";
+import "./utils/import-app.test.ts";
+import "./utils/chai-plugin.test.ts";
```
Comment (takameyer) [inline integration-tests/tests/src/index.ts:31]: Is there a way to detect these extensions automatically?

**Review by takameyer — APPROVED** [review]
Just a few small things, but nothing holding this back.  LGTM

File: integration-tests/tests/src/index.ts (hunk 1 above)
Comment (kraenhansen) [reply integration-tests/tests/src/index.ts:31]: Yeah, I played around with it. It think it gets confused about the `.test` part of the extension 😕

File: integration-tests/environments/node/index.mjs [hunk 2]
```diff
@@ -16,23 +16,34 @@
...
+    global.fs = fs;
+    global.path = path;
     global.environment = { ...context, node: true };
 
-    // Add the integration test suite (in TypeScript)
-    require("ts-node/register/transpile-only");
-    require("@realm/integration-tests");
+    // Add the integration test suite
+    await import("@realm/integration-tests");
+    console.log("integration tests loaded");
```
[the remark this answers is not in the document]
Comment (kraenhansen) [reply integration-tests/environments/node/index.mjs:35]: Oh yeah - good catch. They need to go 🙂

File: integration-tests/tests/src/index.ts (hunk 1 above)
Comment (takameyer) [reply integration-tests/tests/src/index.ts:31]: I was getting this issue as well when I was getting the react-native tests to run again, but in a different section.  Wasn't really able to resolve it.  Very odd.

**Pull request merged**